### ISSUE TITLE
Cannot read property 'defaultMetadataStorage' of undefined

### ISSUE THREAD

**Issue by rico** [typestack/class-transformer#23]
Im using your package in an Ionic2 / Angular2 project. Since the upgrade to the latest RC version of Ionic, where [rollup.js](http://rollupjs.org/) is used as a build system, I get the following error:

```
error_handler.js:51 TypeError: Cannot read property 'defaultMetadataStorage' of undefined
  at TransformOperationExecutor.getKeys (TransformOperationExecutor.js:219)
  at TransformOperationExecutor.transform (TransformOperationExecutor.js:68)
  at ClassTransformer.plainToClass (ClassTransformer.js:16)
  at plainToClass (index.js:20)
```

Now, probability us great, that the build process is causing the problem. But if someone has an idea on how to solve this - please do not hesitate ;-)

**Comment by strictd** [comment]
in my ionic www/build/main.js the TransformOperationExecutor is getting compiled in before the index component (which contains defaultMetadataStorage). TransformOperationExecutor needs the reference from createCommonjsModule(function (module, exports) which is defined after it's needed.

**Comment by strictd** [comment]
I'm sure this issue is a cyclical dependency the rollup is unable to deal with. Going back to rollup 0.34 you get the error messages; I can go into the build/main.js and remap the TransformOperationExecutor, index_1 references to the global variable set by createCommonjsModule (index_1$5 for me) and everything works great.

```
rollup: Module commonjs-proxy:/Users/XXXX/node_modules/class-transformer/index.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 

rollup: Module /Users/XXXX/node_modules/class-transformer/TransformOperationExecutor.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 

rollup: Module commonjs-proxy:/Users/XXXX/node_modules/class-transformer/TransformOperationExecutor.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 

rollup: Module /Users/XXXX/node_modules/class-transformer/ClassTransformer.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 

rollup: Module /Users/XXXX/node_modules/class-transformer/decorators.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 

rollup: Module commonjs-proxy:/Users/XXXX/node_modules/class-transformer/ClassTransformer.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering

rollup: Module commonjs-proxy:/Users/XXXX/node_modules/class-transformer/decorators.js 
may be unable to evaluate without /Users/XXXX/node_modules/class-transformer/index.js, 
but is included first due to a cyclical dependency. Consider swapping the import statements 
in /Users/XXXX/src/mobile/app/main.dev.ts to ensure correct 
ordering 
```

**Comment by strictd** [comment]
fixed, added pull request https://github.com/pleerock/class-transformer/pull/25

**Comment by pleerock** [comment]
thanks @strictd released in 0.1.2
